File: typescipt/src/compiler/JavaScript/builtInFunctions/hourFunctionCall.ts
import {SingleArgumentFunctionCall} from "./singleArgumentFunctionCall";
import {HourFunction} from "../../../language/expressions/functions/hourFunction";

export class HourFunctionCall extends SingleArgumentFunctionCall {
   protected override className = "BuiltInDateFunctions";
   protected override methodName = "hour";

   constructor(functionNode: HourFunction) {
      super(functionNode);
   }
}

File: dotnet/Lexy/Lexy.Compiler/Compiler/CSharp/BuiltInFunctions/HourFunctionCall.cs
using Lexy.Compiler.Language.Expressions.Functions;
using Lexy.RunTime;

namespace Lexy.Compiler.Compiler.CSharp.BuiltInFunctions;

internal class HourFunctionCall : SingleArgumentFunctionCall<HourFunction>
{
    protected override string ClassName => nameof(BuiltInDateFunctions);
    protected override string MethodName => nameof(BuiltInDateFunctions.Hour);
}